os/fs/smartfs/smartfs_smart.c

```diff
@@ -1922,46 +1956,70 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 		mode = oldentry.flags & SMARTFS_DIRENT_MODE;
 		type = oldentry.flags & SMARTFS_DIRENT_TYPE;
 
-		ret = smartfs_createentry(fs, newparentdirsector, newfilename, type, mode, &newentry, oldentry.firstsector, NULL);
-		if (ret != OK) {
-			goto errout_with_semaphore;
-		}
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+		if (oldparentdirsector == newparentdirsector) {
+			ret = smartfs_replaceentry(fs, oldentry, newfilename, type, mode, &newentry, NULL);
+			if (ret != OK) {
+				goto errout_with_semaphore;
+			}
+		} else {
+			ret = smartfs_find_firstinvalidentry(fs, newparentdirsector, &invalidentry);
```

Comment:
Actually I don't like this name, you are good at english, could you please suggest nice name something means that 'overwrite' or .something? (Not this PR, consider it when you have some time)

Reply:
Ohkkay, I'll give this a thoight.